alicloud/resource_alicloud_alikafka_instance.go

```diff
@@ -0,0 +1,332 @@
+	upgradeReq.RegionId = client.RegionId
+	upgradeReq.InstanceId = d.Id()
+	upgradeReq.TopicQuota = requests.NewInteger(d.Get("topic_quota").(int))
+	upgradeReq.DiskSize = requests.NewInteger(d.Get("disk_size").(int))
+	upgradeReq.IoMax = requests.NewInteger(d.Get("io_max").(int))
+
+	if d.HasChange("topic_quota") || d.HasChange("disk_size") || d.HasChange("io_max") {
+		attributeUpdate = true
+	}
+	eipMax := 0
+	if v, ok := d.GetOk("eip_max"); ok {
+		eipMax = v.(int)
+	}
+	if d.HasChange("eip_max") {
+
+		if v, ok := d.GetOk("eip_max"); ok {
+			eipMax = v.(int)
+		}
+		upgradeReq.EipMax = requests.NewInteger(eipMax)
+		attributeUpdate = true
+	}
+
+	if attributeUpdate {
+		err := resource.Retry(5*time.Minute, func() *resource.RetryError {
+			raw, err := alikafkaService.client.WithAlikafkaClient(func(alikafkaClient *alikafka.Client) (interface{}, error) {
+				return alikafkaClient.UpgradePostPayOrder(upgradeReq)
+			})
+			if err != nil {
+				if IsExceptedErrors(err, []string{AlikafkaThrottlingUser}) {
+					time.Sleep(10 * time.Second)
+					return resource.RetryableError(err)
+				}
+				return resource.NonRetryableError(err)
+			}
+			addDebug(upgradeReq.GetActionName(), raw, upgradeReq.RpcRequest, upgradeReq)
+			return nil
+		})
+		if err != nil {
+			return WrapErrorf(err, DefaultErrorMsg, d.Id(), upgradeReq.GetActionName(), AlibabaCloudSdkGoERROR)
+		}
+	}
+
+	err := alikafkaService.WaitForAlikafkaInstanceUpdated(d.Id(), d.Get("topic_quota").(int),
+		d.Get("disk_size").(int), d.Get("io_max").(int), eipMax, DefaultTimeoutMedium)
+
+	if err != nil {
+		return WrapError(err)
+	}
+
+	err = alikafkaService.WaitForAlikafkaInstance(d.Id(), Running, 2000)
+
+	if err != nil {
+		return WrapError(err)
+	}
+
+	return resourceAlicloudAlikafkaInstanceRead(d, meta)
+}
+
+func resourceAlicloudAlikafkaInstanceDelete(d *schema.ResourceData, meta interface{}) error {
+
+	client := meta.(*connectivity.AliyunClient)
+	alikafkaService := AlikafkaService{client}
+
+	request := alikafka.CreateReleaseInstanceRequest()
+	request.InstanceId = d.Id()
+	request.RegionId = client.RegionId
+	request.ReleaseIgnoreTime = requests.NewBoolean(true)
+	request.ForceDeleteInstance = requests.NewBoolean(true)
+
+	err := resource.Retry(5*time.Minute, func() *resource.RetryError {
+		raw, err := alikafkaService.client.WithAlikafkaClient(func(alikafkaClient *alikafka.Client) (interface{}, error) {
+			return alikafkaClient.ReleaseInstance(request)
+		})
+		if err != nil {
+			if IsExceptedErrors(err, []string{AlikafkaThrottlingUser}) {
+				time.Sleep(10 * time.Second)
+				return resource.RetryableError(err)
+			}
+			return resource.NonRetryableError(err)
+		}
+		addDebug(request.GetActionName(), raw, request.RpcRequest, request)
+		return nil
+	})
+	if err != nil {
+		return WrapErrorf(err, DefaultErrorMsg, d.Id(), request.GetActionName(), AlibabaCloudSdkGoERROR)
+	}
+
+	return WrapError(alikafkaService.WaitForAllAlikafkaNodeRelease(d.Id(), DefaultTimeoutMedium))
```

Comment:
please use timeouts to do it, like https://github.com/terraform-providers/terraform-provider-alicloud/blob/master/alicloud/resource_alicloud_instance.go#L779